e3sm_diags/driver/aerosol_budget_driver.py

```diff
@@ -110,94 +46,225 @@ def generate_metrics_dic(data, aerosol, season):
+    sink_gi = global_integral((drydep - wetdep), area_m2) * UNITS_CONV
+    drydep_gi = global_integral(drydep, area_m2) * UNITS_CONV
+    wetdep_gi = global_integral(wetdep, area_m2) * UNITS_CONV
+
+    if aerosol in ["bc", "pom", "so4"]:
+        var_key = f"{aerosol}_CLXF"
+        elvemis = ds_test.get_climo_dataset(var_key, season)[var_key]
+
+        elvemis_gi = global_integral(elvemis, area_m2) * UNITS_CONV
+    else:
+        elvemis_gi = 0.0
+
+    metrics_dict: MetricsDict = {
+        "Surface Emission (Tg/yr)": f"{srfemis_gi:.2f}",
+        "Elevated Emission (Tg/yr)": f"{elvemis_gi:.2f}",
+        "Sink (Tg/yr)": f"{sink_gi:.2f}",
+        "Dry Deposition (Tg/yr)": f"{drydep_gi:.2f}",
+        "Wet Deposition (Tg/yr)": f"{wetdep_gi:.2f}",
+        "Burden (Tg)": f"{burden_total:.2f}",
+        "Lifetime (Days)": f"{burden_total/sink_gi*365:.2f}",
+    }
+
+    return metrics_dict
+
+
+def _calc_column_integral(
+    test_ds: Dataset, aerosol: str, season: ClimoFreq
+) -> xr.DataArray:
+    """Calculate column integrated mass
+
+    Take aerosol and change it to the appropriate string.
+      - ncl -> SEASALT, dst -> DUST, rest1 -> REST1
+      - only needed if ABURDEN terms are available
+
+    Parameters
+    ----------
+    test_ds : Dataset
+        The e3sm_diags test dataset object.
+    aerosol : str
+        The aerosol key.
+    season : ClimoFreq.
+        The climo season.
+
+    Returns
+    -------
+    xr.DataArray
+        The column integrated mass.
+    """
+    if aerosol == "ncl":
+        aerosol_name = "SEASALT"
+    elif aerosol == "dst":
+        aerosol_name = "DUST"
+    else:
+        aerosol_name = aerosol.upper()
+
+    try:
+        # if ABURDEN terms are available, use them
+        var_key = f"ABURDEN{aerosol_name}"
+        burden = test_ds.get_climo_dataset(var_key, season)[var_key]
+    except IOError:
+        # if not, use the Mass_ terms and integrate over the column
+        var_key = f"Mass_{aerosol}"
+        ds_mass = test_ds.get_climo_dataset(var_key, season)
+        mass = ds_mass[var_key]
+
+        pressure_levs = _hybrid_to_pressure(
+            ds_mass, var_key, p0=100000.0, a_key="hyai", b_key="hybi"
+        )
+        # Preserve the original units.
+        pressure_levs.attrs["units"] = ds_mass[var_key].attrs["units"]
+
+        # (72,lat,lon)
+        plev_key = xc.get_dim_keys(pressure_levs, axis="Z")
+        delta_p = pressure_levs.diff(plev_key)
+
+        # Perform numpy index-based arithmetic instead of xarray label-based
+        # arithmetic because the Z dims of `mass` and `delta_p` can have
+        # different names ("lev" vs. "ilev") with slightly different values. The
+        # CDAT version of this code uses numpy index-based arithmetic too.
+        with xr.set_options(keep_attrs=True):
+            # mass density * mass air   kg/m2
+            mass_3d = mass.copy()
+            mass_3d.values = mass.values * delta_p.values / 9.8
```

Comment:
We can align the Z axis dim names then use [xr.align() ](https://docs.xarray.dev/en/stable/generated/xarray.align.html) to get the Z axis coordinates aligned before performing label-based arithmetic.